pkg/proxyagent/agent/agent.go

```diff
@@ -231,7 +294,83 @@ func CustomSignerWithExpiry(customSignerName string, caKey, caData []byte, durat
 
 const (
 	ApiserverNetworkProxyLabelAddon = "open-cluster-management.io/addon"
-
-	AgentSecretName   = "cluster-proxy-open-cluster-management.io-proxy-agent-signer-client-cert"
-	AgentCASecretName = "cluster-proxy-ca"
+	AgentSecretName                 = "cluster-proxy-open-cluster-management.io-proxy-agent-signer-client-cert"
+	AgentCASecretName               = "cluster-proxy-ca"
 )
+
+func filterMCS(managedClusterSets []clusterv1beta1.ManagedClusterSet, clusterlabels map[string]string) ([]clusterv1beta1.ManagedClusterSet, error) {
+	filteredManagedClusterSets := []clusterv1beta1.ManagedClusterSet{}
+	for i := range managedClusterSets {
+		mcs := managedClusterSets[i]
+
+		// deleted managedClusterSets are not included in the list
+		if !mcs.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		// only cluseterSet cover current cluster include in the list.
+		selector, err := clusterv1beta1.BuildClusterSelector(&mcs)
+		if err != nil {
+			return nil, err
+		}
+		if !selector.Matches(labels.Set(clusterlabels)) {
+			continue
+		}
+
+		filteredManagedClusterSets = append(filteredManagedClusterSets, mcs)
+	}
+	return filteredManagedClusterSets, nil
+}
+
+func filterMPSR(serviceResolvers []proxyv1alpha1.ManagedProxyServiceResolver, managedClusterSetMap map[string]clusterv1beta1.ManagedClusterSet) []proxyv1alpha1.ManagedProxyServiceResolver {
```

Comment:
how about return the serviceToExposes directly

Reply:
Done.